product/tutorials/tutorial_add_fulfillment_places.md

```diff
@@ -0,0 +1,147 @@
+
+To do that, run the following commands in a terminal:
+
+```bash
+python3 -m venv tutorial-env
+```
+
+```bash
+source tutorial-env/bin/activate
+```
+
+Next, install Google packages:
+
+```bash
+python3 -m pip install google
+```
+
+```bash
+python3 -m pip install google.cloud.retail
+```
+
+**Tip**: Click the copy button on the side of the code box to paste the command in the Cloud Shell terminal and run it.
+
+## Set the PROJECT_NUMBER environment variable
+
+As you are going to run the code samples in your own Cloud Project, you should specify the **project_id** as an environment variable, it will be used in every request to the Retail API.
+
+You can find the ```project_number``` in the **Home/Dashboard/Project Info card**.
+
+Set the environment variable with a following command:
+```bash
+export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+```
+
+## Add fulfillment places 
+
+To send the AddFulfillmentPlacesRequest you need to set the following fields:
+ - ```product``` - the product name which inventory information you wnd to update,
+ - ```type``` - the fulfillment type, you may set commonly used types (such as pickup in store and same day delivery) or custom types,
+ - ```place_ids[]``` - the store IDs for each of the fulfillment type,
+ - ```add_time``` - the time when the fulfillment updates are issued, used to prevent out-of-order updates on fulfillment information. If not provided, the internal system time will be used,
+ - ```allow_missing``` - if set to true, and the Product is not found, the fulfillment information will still be processed and retained for at most 1 day and processed once the Product is created. 
+
+Open **product/add_remove_fulfillment.py** and check the AddFulfillmentPlacesRequest.
+
+To add the fulfillment places, open terminal and run the command:
+
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the responses in the Terminal. As you can see, the product was initially created without the fulfillment information, 
+then check the get_product() response, the fulfillment places 'store1', 'store2', 'store3' was added to the fulfillment type 'pickup-in-store'.
+
+## Send out-of-order add_fulfillment_places request
+
+The AddFulfillmentPlaces method allows the caller to specify an update time for when the request is issued.
+This update time is compared against the latest update time recorded for the relevant inventory fields,
+and the update is committed if and only if the update time is strictly after the latest update time.
+
+Let's modify the ```add_fulfillment_request```, change the list of the **place_ids**.
+
+Change the ```request_time``` value, set now() - 1 day:
+```
+request_time = datetime.datetime.now() - datetime.timedelta(days=1)
+```
+
+Run the code once again:
+```bash
+python product/add_remove_fulfillment.py
+```
+
+Check the product printed out in the Terminal, the fulfillment places information was not updated.
```

Comment:
Check the product printed out in the Terminal. The fulfillment places information has no updates.